packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -22,6 +23,20 @@ export class LanguageServerResolver {
         private readonly _defaultDownloadFolder?: string
     ) {}
 
+    // wraps the resolver to show download status message
+    async resolveWithProgress() {
```

Comment:
Not clear what this is intended to do, but I'll just mention that we have a pretty sophisticated progress wrapper: `showProgressWithTimeout` https://github.com/aws/aws-toolkit-vscode/blob/30face99ec161876378b49e71e78ae3714ed5797/packages/core/src/shared/utilities/messages.ts#L239

which also supports `Cancel`. The ability to cancel is very important for most "progress" tasks.